drivers/usb/udc/udc_dwc2.c

```diff
@@ -130,9 +133,11 @@ struct udc_dwc2_data {
 };
 
 static void dwc2_on_bus_reset(const struct device *dev);
-static void dwc2_restore_device_registers(const struct device *dev);
+static void dwc2_restore_device_registers(const struct device *dev,
+					  bool remote_wakeup);
 static void dwc2_enter_hibernation(const struct device *dev);
-static void dwc2_exit_hibernation(const struct device *dev);
+static void dwc2_exit_hibernation(const struct device *dev, bool remote_wakeup);
+static void cancel_hibernation_request(struct udc_dwc2_data *const priv);
```

Comment:
Please get rid of terrible forward definitions.

Reply:
Added new commit reordering the functions prior to all the changes here.